### PR TITLE
DIOS-7446 Avoid calling pthread_kill if thread is not running

### PR DESCRIPTION
**Pull request by harryz2000**

### CODE REVIEW

**Review by murillo128 — APPROVED** [review]
lgtm

File: src/rtmp/rtmpconnection.cpp [hunk 1]
```diff
@@ -311,7 +311,8 @@ void RTMPConnection::SignalWriteNeeded()
 	pthread_mutex_unlock(&mutex);
 
 	//Signal the pthread this will cause the poll call to exit
-	pthread_kill(thread.native_handle(),SIGIO);
+	if (running)
```
Comment (craig-johnston) [inline src/rtmp/rtmpconnection.cpp:314]: I don't know this code, so general question:
Is there any chance of a race condition here? Can the thread stop running while this code (the next line specifically) is executing?
The variable running isn't being read under a mutex.

**Review by craig-johnston — CHANGES_REQUESTED** [review]

File: src/rtmp/rtmpconnection.cpp (hunk 1 above)
Comment (harryz2000) [reply src/rtmp/rtmpconnection.cpp:314]: Yeah. Ideally, we would use EventLoop which has the eventfd to break the poll wait. This is a hot fix.

Comment (craig-johnston) [reply src/rtmp/rtmpconnection.cpp:314]: It'll definitely crash a lot less and if the better fix is coming later, all good.

**Pull request merged**